### PR TITLE
fix: session update and company form submitting issue

### LINKED ISSUE #174: Switching between the company seems to be broken
switching-company

### PR DESCRIPTION
**Pull request by anikdhabal**
fixes #174 

Currently, there are some bugs present."

1.The company form is not submitting because there was an optional field present in the form, and we conditionally render those fields. Since those fields are optional, they will only be typeof string in our case, or undefined when not rendered. However, we set the default values to string and also conditionally not render those fields.

2.It doesn't allow you to switch the company or navigate to the dashboard page after completing the onboarding. This is because of the 'if' block; every time it returns the new token when you update the session.

Screenshot 2024-03-03 201250

And we don't need to explicitly write those blocks, as we update the session directly when updating the user profile.

Screenshot 2024-03-03 201352

### CODE REVIEW

**Review by G3root — CHANGES_REQUESTED** [review]
Great find @anikdhabal ❤️ . currently blocking until #167 is figured out because it touches some code in company-form.

File: src/server/auth.ts [hunk 1]
```diff
@@ -73,17 +73,7 @@ export const authOptions: NextAuthOptions = {
...
-      if (trigger === "update") {
-        const newToken = {
-          ...token,
-          ...session?.user,
-          picture: session?.user?.image || "",
-        };
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-        return newToken;
-      }
-
```
Comment (G3root) [inline src/server/auth.ts:86]: we have to pass the profile picture url through jwt callback and session callback.

**Comment by anikdhabal** [comment]
> Great find @anikdhabal ❤️ . currently blocking until #167 is figured out because it touches some code in company-form.

Thanks Nafeez for the review.

**Review by dahal — CHANGES_REQUESTED** [review]
@anikdhabal @G3root #167 PR is merged. Please resolve the merge conflict and make necessary changes. It looks good to me and huge props to @anikdhabal for identifying the issue and taking the initiative to resolve it.

**Review by dahal — CHANGES_REQUESTED** [review]
Company create or update flow seems to be broken? Please take a look at this, its not sending a TRPC request.


company-create-update-flow

**Comment by anikdhabal** [comment]
> Company create or update flow seems to be broken? Please take a look at this, its not sending a TRPC request.
> 
> company-create-update-flow [ company-create-update-flow ](https://private-user-images.githubusercontent.com/3684236/310745460-8ce30d9f-47fd-4d03-89be-2ff51624f12e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3ODYwNTgsIm5iZiI6MTcwOTc4NTc1OCwicGF0aCI6Ii8zNjg0MjM2LzMxMDc0NTQ2MC04Y2UzMGQ5Zi00N2ZkLTRkMDMtODliZS0yZmY1MTYyNGYxMmUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDdUMDQyOTE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTA1ZDA1MDYzMWZlNzQ3YTRiNDc1ZDNjNjM3OGY5OTNjMGQwNjZmYWExZmE2YjVjNGI5NWY0N2Q0M2Q4OGI1YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aHuf96tlvnfpCgcZjrLtiWmEFKEdRGATd9b8OT-zo7o) [ ](https://private-user-images.githubusercontent.com/3684236/310745460-8ce30d9f-47fd-4d03-89be-2ff51624f12e.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3ODYwNTgsIm5iZiI6MTcwOTc4NTc1OCwicGF0aCI6Ii8zNjg0MjM2LzMxMDc0NTQ2MC04Y2UzMGQ5Zi00N2ZkLTRkMDMtODliZS0yZmY1MTYyNGYxMmUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMDdUMDQyOTE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTA1ZDA1MDYzMWZlNzQ3YTRiNDc1ZDNjNjM3OGY5OTNjMGQwNjZmYWExZmE2YjVjNGI5NWY0N2Q0M2Q4OGI1YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.aHuf96tlvnfpCgcZjrLtiWmEFKEdRGATd9b8OT-zo7o)

Yeah, this PR fixed that issue as well. I also explained in detail why the form is not submitting.

**Comment by dahal** [comment]
> > Company create or update flow seems to be broken? Please take a look at this, its not sending a TRPC request.
> 
> Yeah, this PR fixed that issue as well. I also explained in detail why the form is not submitting.

I was testing it on your current PR, also made sure I have the latest commits, still the same issue.

**Comment by anikdhabal** [comment]
> > > Company create or update flow seems to be broken? Please take a look at this, its not sending a TRPC request.
> > 
> > 
> > Yeah, this PR fixed that issue as well. I also explained in detail why the form is not submitting.
> 
> I was testing it on your current PR, also made sure I have the latest commits, still the same issue.

Thanks puru, I've updated the code. Now Everything is Ok.

**Pull request merged**